File: Source/Kernel/Storage.Specs/EventSequences/for_HistogramResolutions/when_picking_a_resolution_for_a_span.cs
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Cratis.Chronicle.Storage.EventSequences.for_HistogramResolutions;

public class when_picking_a_resolution_for_a_span
{
    [Fact] void should_pick_minutes_for_an_hour() =>
        HistogramResolutions.ForSpan(TimeSpan.FromHours(1)).ShouldEqual(HistogramResolution.Minute);

    [Fact] void should_pick_hours_for_a_day() =>
        HistogramResolutions.ForSpan(TimeSpan.FromDays(1)).ShouldEqual(HistogramResolution.Hour);

    [Fact] void should_pick_days_for_a_month() =>
        HistogramResolutions.ForSpan(TimeSpan.FromDays(30)).ShouldEqual(HistogramResolution.Day);

    [Fact] void should_pick_weeks_for_a_year() =>
        HistogramResolutions.ForSpan(TimeSpan.FromDays(365)).ShouldEqual(HistogramResolution.Week);

    [Fact] void should_pick_months_for_a_decade() =>
        HistogramResolutions.ForSpan(TimeSpan.FromDays(3650)).ShouldEqual(HistogramResolution.Month);

    [Fact] void should_pick_minutes_for_an_empty_span() =>
        HistogramResolutions.ForSpan(TimeSpan.Zero).ShouldEqual(HistogramResolution.Minute);
}

File: Source/Workbench/Features/EventStore/Namespaces/Sequences/QueryEditor/for_histogramResolution/when_picking_a_resolution_for_a_span.ts
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

import { resolutionForSpan } from '../histogramResolution';

const minute = 60 * 1000;
const hour = 60 * minute;
const day = 24 * hour;

describe('when picking a resolution for a span', () => {
    it('should use minutes for an hour', () => resolutionForSpan(hour).should.equal('minute'));
    it('should use hours for a day', () => resolutionForSpan(day).should.equal('hour'));
    it('should use days for a month', () => resolutionForSpan(30 * day).should.equal('day'));
    it('should use weeks for a year', () => resolutionForSpan(365 * day).should.equal('week'));
    it('should use months for a decade', () => resolutionForSpan(3650 * day).should.equal('month'));
    it('should use minutes for an empty span', () => resolutionForSpan(0).should.equal('minute'));
});
